### PR TITLE
docs: Update component structure and storybook documentation

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
- Updates our component structure
- Adds modern example
- Clarifies how to name interface/type `Props`
- Removes old way of writing stories with recommended format for Storybook 7.0

### CODE REVIEW

File: docs/development-guide/02-component-structure.md [hunk 1]
```diff
@@ -6,41 +6,46 @@ The basic structure of a component file should follow:
...
-  name: string;
+interface SayHelloProps {
+    name: string;
+    isDisabled: boolean;
 }
 
-const SayHello: React.FC<Props> = (props) => {
-  const classes = useStyles();
-  return <h1 className={classes.name}>Hello, {capitalize(props.name)}</h1>;
+const SayHello = ({ name, isDisabled }: SayHelloProps) => {
```
Comment (mjac0bs) [inline docs/development-guide/02-component-structure.md:28]: I think I still prefer
```
const { name, isDisabled } = SayHelloProps;
```
for consistency and especially readability when we're destructuring a longer list of props.

File: docs/development-guide/04-component-library.md [hunk 2]
```diff
@@ -47,55 +47,8 @@ A composition of Core Styles and Elements. Normally with some code that defines
...
-    args={{
-      open: true,
-      typeToConfirm: true,
-      entity: 'Linode',
-      label: 'my-linode-0',...>
-```
-
 #### Best Practices
+Our stories are in the process of being updated to the latest Storybook 7.0 format.
+We currently use MDX both for documentation and for defining stories in the same .stories.mdx file. However, Storybook has deprecated this functionality and plan to remove it in a future version of Storybook.
```
Comment (mjac0bs) [inline docs/development-guide/04-component-library.md:52]:
```suggestion
We currently use MDX both for documentation and for defining stories in the same `.stories.mdx` file. However, Storybook has deprecated this functionality and we plan to remove it in a future version of Storybook.
```

File: docs/development-guide/04-component-library.md [hunk 3]
```diff
@@ -47,55 +47,8 @@ A composition of Core Styles and Elements. Normally with some code that defines
...
-  - From a decorator, make sure you return `Story(props)` instead of `<Story {…props}/>` or else you’ll see `<No Display Name />` in the code preview tab.
-  - To access given props from your template, use the [second “context” argument](https://github.com/linode/manager/pull/8060/files#diff-c7a7ccf5bc5311daaff07fd6fd33a40e4e73a06b4df8a337f9ce94f0577fdceeR13) in the template function.
-- Mocking data/API calls can be tricky, and the solution may be component-dependent.
-- MDX is unforgiving. Check the browser console for Storybook errors with syntax, imports, etc.
-- Use the recommended Markdown and MDX VSCode plugins outlined in our [Coding Standards](./13-coding-standards.md).
-- Avoid using the `useArgs` hook. It should only be used when there needs to be a two way binding between component state and the values shown in the `ArgsTable`.
-- When defining an `ArgsTable` always set the `sort` prop to `requiredFirst`.
+As we begin to move away from the MDX formatPlease refer to their [documentation](https://storybook.js.org/docs/react/writing-docs/introduction) for how to write stories in the CSF format.
```
Comment (mjac0bs) [inline docs/development-guide/04-component-library.md:54]:
```suggestion
As we begin to move away from the MDX format, please refer to Storybook's  [documentation](https://storybook.js.org/docs/react/writing-docs/introduction) for how to write stories in the CSF format.
```

File: docs/development-guide/02-component-structure.md
Comment (mjac0bs) [inline docs/development-guide/02-component-structure.md:1]: Looks like we'll want to remove this part on default exports: https://github.com/linode/manager/pull/8969/files#diff-56dccdb618dbd906ae625af55dcaa80ec1510297366a6aac59fd75089bc5871eR81-R84
```
#### Default Export
- Usually you'll export the component by default,
  - e.g. `export default = MyComponent`
```

**Review by mjac0bs — APPROVED** [review]
🚀

**Review by cpathipa — APPROVED** [review]

**Pull request merged**